src/app/core/i18n/i18n.service.ts

```diff
@@ -5,41 +5,46 @@ import {loadTranslations} from "@angular/localize";
 @Injectable({
     providedIn: "root",
   })
-  export class I18n {
-    locale = "en";
-    readonly allowedLocales = ["en", "fr", "de", "it"];
+  export class I18nService {
+    readonly allowedLanguages = ["en", "fr", "de", "it"];
     translations: any = {};
-  
-    async setLocale() {
-      const userLocale = localStorage.getItem("locale");
-  
-      // If the user has a preferred language stored in LocalStorage, use it.
-      if (userLocale && this.allowedLocales.includes(userLocale)) {
-        this.locale = userLocale;
-      } else {
-        localStorage.setItem("locale", this.locale);
+    language: string;
+
+    async setLanguage() {
```

Comment:
Hm, it's a bit awkward to make it the caller's responsibility to set the local storage item, just so that this function can fetch it again. Could we perhaps make it so `setLanguage()` takes an optional `language` argument, and all of the local storage related shenanigans stay inside the i18n service?

Reply:
Done, should be ok now